admin/yum.py

```diff
@@ -93,18 +101,42 @@ class CreateRepo(object):
 def perform_create_repository(dispatcher, intent):
     """
     See :class:`CreateRepo`.
-
-    :return: List of new and modified rpm metadata filenames.
     """
-    # The update option means that this is faster when there is existing
-    # metadata but has output starting "Could not find valid repo at:" when
-    # there is not existing valid metadata.
-    check_call([
-        b'createrepo',
-        b'--update',
-        b'--quiet',
-        intent.repository_path.path])
-    return _list_new_metadata(repository_path=intent.repository_path)
+    from .packaging import PackageTypes, Distribution
+
+    distribution = Distribution(
+        name=intent.distro_name,
+        version=intent.distro_version,
+    )
+    package_type = distribution.package_type()
+
+    if package_type == PackageTypes.RPM:
+        # The update option means that this is faster when there is existing
+        # metadata but has output starting "Could not find valid repo at:" when
+        # there is not existing valid metadata.
+        check_call([
+            b'createrepo',
+            b'--update',
+            b'--quiet',
+            intent.repository_path.path])
+        return _list_new_metadata(repository_path=intent.repository_path)
+    elif package_type == PackageTypes.DEB:
```

Comment:
I had started to make this change, and got part way through and it was getting large, and slightly confusing. It would be much better as a separate follow-up branch.